implementation/rest/src/main/java/io/smallrye/opentelemetry/implementation/rest/OpenTelemetryServerFilter.java

```diff
@@ -142,7 +151,15 @@ protected InetSocketAddress getPeerSocketAddress(final ContainerRequestContext r
         @Override
         protected InetSocketAddress getHostSocketAddress(final ContainerRequestContext request) {
             URI requestUri = request.getUriInfo().getRequestUri();
-            return new InetSocketAddress(requestUri.getHost(), requestUri.getPort());
+            if (requestUri.getPort() > 0) {
+                return new InetSocketAddress(requestUri.getHost(), requestUri.getPort());
+            }
+            try {
+                return new InetSocketAddress(requestUri.getHost(), requestUri.toURL().getDefaultPort());
+            } catch (MalformedURLException ex) {
+                return new InetSocketAddress(requestUri.getHost(), requestUri.getPort());
+            }
+
```

Comment:
Could this simply be `return new InetSocketAddress(requestUri.getHost(), hostPort(request));`

(Or extract the shared logic into a method that takes `URI` if there's concern about the cost of `request.getUriInfo().getRequestUri()`?)

Reply:
yes it can